pkg/controllers/provisioning/provisioner.go

```diff
@@ -365,9 +365,17 @@ func (p *Provisioner) getDaemonSetPods(ctx context.Context) ([]*v1.Pod, error) {
 	if err := p.kubeClient.List(ctx, daemonSetList); err != nil {
 		return nil, fmt.Errorf("listing daemonsets, %w", err)
 	}
+
 	var ret []*v1.Pod
-	for _, daemonSet := range daemonSetList.Items {
-		pod := &v1.Pod{Spec: daemonSet.Spec.Template.Spec}
+	for index := range daemonSetList.Items {
+		cachedPod := p.cluster.GetDaemonSetPod(&daemonSetList.Items[index])
+		var pod *v1.Pod
```

Comment:
I think this can all be simplified down to 

```go
pod := p.cluster.GetDaemonSetPod(&daemonSetList.Items[i])
if pod == nil {
  pod = &v1.Pod{Spec: daemonSetList.Items[i].Spec.Template.Spec}
}
```

Reply:
Even better

```go
pods := lo.Map(daemonSetList.Items, func(d appsv1.DaemonSet, _ int) *v1.Pod {
  pod := p.cluster.GetDaemonSetPod(d)
  if pod == nil {
    pod = &v1.Pod{Spec: d.Spec.Template.Spec}
  }
  return pod
})
```